pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -184,6 +192,18 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       requestStatistics.setErrorCode(QueryException.PQL_PARSING_ERROR_CODE);
       return new BrokerResponseNative(QueryException.getException(QueryException.PQL_PARSING_ERROR, e));
     }
+    if (isLiteralOnlyQuery(brokerRequest)) {
+      LOGGER.info("Request {} contains only Literal, skipping server query: {}", requestId, query);
+      try {
+        BrokerResponse brokerResponse =
+            processLiteralOnlyBrokerRequest(brokerRequest, compilationStartTimeNs, requestStatistics);
+        return brokerResponse;
+      } catch (IllegalStateException e) {
```

Comment:
Why only catching `IllegalStateException`?

Reply:
We only catch the behavior that process literal functions failure.

Reply:
IMO we should catch all `Exception` and fallback to the old behavior instead of directly failing the query. What if `processLiteralOnlyBrokerRequest()` somehow throws other type of exceptions?

Reply:
I see. My purpose here is to prevent broken query pushing down to servers for further processing. E.g. Function AS has two arguments first is any expression, second is an identifier, however if it's not, then typically it means the query is not writing correct.

I'll change it to catch all exceptions and add a TODO for further refinements.